### PR TITLE
Kep for node resources allocatable score plugin

### PR DESCRIPTION
**Pull request by patricklundquist**
Here's the kep for node resources allocatable score plugin as requested by @Huang-Wei in comments of #36 

/assign @Huang-Wei

### CODE REVIEW

File: kep/20200903-node-resources-allocatable-scoring.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
```
Comment (Huang-Wei) [inline kep/20200903-node-resources-allocatable-scoring.md:1]: Let's rename this file into `kep/48-node-resources-allocatable-scoring/README.md`, and move the metadata info (L1 to L15) to `kep/48-node-resources-allocatable-scoring/kep.yaml`.

The latest examples can be checked out under https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/kep.

File: kep/20200903-node-resources-allocatable-scoring.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+## Terms
+
+## Proposal
+
+In order to implement node resources allocatable scoring, we developed a score plugin. The plugin takes an arg ```mode```, which can have values ```Least``` or ```Most```. When ```Least``` is specified nodes with the least allocatable resources are scored higher. When ```Most``` is specified, nodes with the most allocatable resources are scored higher. ```Least``` and ```Most``` are totally opposite semantics, so it's only possible to have one enabled in a scheduler profile. The plugin takes another arg ```resources``` to specify the ```weight``` and ```name``` of each resource used in the scoring calculations.
+
+## Design Details
+We implement a score plugin based on the scheduler framework. Implementation follows a similar pattern to the existing ```NodeResourcesLeastAllocated``` score plugin.
+
```
Comment (Huang-Wei) [inline kep/20200903-node-resources-allocatable-scoring.md:63]: It'd be good to give a concrete example explaining how the score is calculated.

File: kep/20200903-node-resources-allocatable-scoring.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  - [Alternatives considered](#alternatives-considered)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Testing Plan](#testing-plan)
+  - [Implementation History](#implementation-history)
+  - [References](#references)
+<!-- /toc -->
+
+## Motivation
+A Kubernetes cluster can consist of nodes with widely varying resource capacities. Likewise, workloads may consist of pods with widely varying resource requests. A pod with high resource requests might only fit on one of the largest capacity nodes. If one or more small pods are scheduled on the largest capacity nodes, the large pod may be unscheduleable. That is a sub-optimal scheduling when there are smaller nodes available that could have taken the small pods. The existing scheduling priorities don't offer a solution to this scenario. To improve cluster utilization, this proposal tries to solve the problem by implementing a scoring plugin ```NodeResourcesAllocatable``` based on the scheduler framework.
+
```
Comment (Huang-Wei) [inline kep/20200903-node-resources-allocatable-scoring.md:42]: We may add a few sentences to explain the shortage of existing `NodeResourcesLeastAllocated` and `NodeResourcesMostAllocated` - which uses **ratio** of allocated resources. While in this plugin, if I understanding it correctly, it's using the **absolute number** of allocatable resources to score/normalize, which quite varies with the in-tree Score plugins.

**Review by Huang-Wei — COMMENTED** [review]
Sry for the delay @patricklundquist. Some comments below.

File: kep/20200903-node-resources-allocatable-scoring.md (hunk 2 above)
Comment (patricklundquist) [reply kep/20200903-node-resources-allocatable-scoring.md]: Done

File: kep/20200903-node-resources-allocatable-scoring.md (hunk 1 above)
Comment (patricklundquist) [reply kep/20200903-node-resources-allocatable-scoring.md]: Done

File: kep/20200903-node-resources-allocatable-scoring.md (hunk 3 above)
Comment (patricklundquist) [reply kep/20200903-node-resources-allocatable-scoring.md]: Done

File: kep/48-node-resources-allocatable-scoring/kep.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+authors:
+  - "@patricklundquist"
+owning-sig: sig-scheduling
+reviewers:
+  - "@Huang-Wei"
+  - "@ahg-g"
+approvers:
+  - "@Huang-Wei"
+creation-date: 2020-09-03
+last-updated: 2020-09-03
```
Comment (Huang-Wei) [inline kep/48-node-resources-allocatable-scoring/kep.yaml:11]:
```suggestion
last-updated: 2020-09-15
```

File: kep/48-node-resources-allocatable-scoring/kep.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+status: provisional
```
Comment (Huang-Wei) [inline kep/48-node-resources-allocatable-scoring/kep.yaml:12]:
```suggestion
status: implementable
```

File: kep/48-node-resources-allocatable-scoring/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+* 2 nodes with allocated/allocatable of 10/20 ( 1 pod: size 10), and 90/100 (2 pods: size 50 and 40)
+* 1 pod of size 10 comes, pod of size 40 is terminated, a new pod of size 50 is scheduled.
+
+With the ```NodeResourcesLeastAllocatable``` score plugin, all pods are scheduled. With other plugins, the size 10 pod may have been scheduled on the larger node and blocked the subsequent size 50 pod.
+
+### Extension points
+
+#### Score
+To score a node we take the sum of it's weighted allocatable resources. The resources and weights to use are user configurable via the plugin args. If the mode is ```least```, then we use the negation of the weighted sum. After all nodes are scored, the scores are normalized to a fixed range defined by the min and max node score in the framework (e.g. [0-100]).
```
Comment (Huang-Wei) [inline kep/48-node-resources-allocatable-scoring/README.md:69]: > If the mode is `least`

I suppose we're gonna use the uppercase `Least`, right?

**Review by Huang-Wei — COMMENTED** [review]
Just some nits, otherwise look pretty neat! Please squash the commits along with addressing the nits.

File: kep/48-node-resources-allocatable-scoring/README.md (hunk 6 above)
Comment (patricklundquist) [reply kep/48-node-resources-allocatable-scoring/README.md]: done

**Pull request merged**